File: CS/Scripts/Syntax/StringExtensions.cs
using System;

namespace ScriptHelp.Scripts.Syntax
{
    /// <summary>
    /// StringExtensions
    /// </summary>
    public static class StringExtensions
    {
        /// <summary>
        /// NormalizeLineBreaks
        /// </summary>
        /// <param name="instance">instance</param>
        /// <param name="preferredLineBreak">preferredLineBreak</param>
        /// <returns>returns the preferred way of doing line breaks</returns>
        public static string NormalizeLineBreaks(this string instance, string preferredLineBreak)
        {
            return instance.Replace("\r\n", "\n").Replace("\r", "\n").Replace("\n", preferredLineBreak);
        }

        /// <summary>
        /// NormalizeLineBreaks
        /// </summary>
        /// <param name="instance">instance</param>
        /// <returns>returns the line break with Environment.NewLine</returns>
        public static string NormalizeLineBreaks(this string instance)
        {
            return NormalizeLineBreaks(instance, Environment.NewLine);
        }
    }
}

File: VB/Scripts/Syntax/StringExtensions.vb

Namespace Scripts
    Namespace Syntax

        Public NotInheritable Class StringExtensions
            Private Sub New()
            End Sub

            Public Shared Function NormalizeLineBreaks(instance As String, preferredLineBreak As String) As String
                Return instance.Replace(vbCr & vbLf, vbLf).Replace(vbCr, vbLf).Replace(vbLf, preferredLineBreak)
            End Function

            Public Shared Function NormalizeLineBreaks(instance As String) As String
                Return NormalizeLineBreaks(instance, Environment.NewLine)
            End Function

        End Class

    End Namespace
End Namespace